Five rules for onboarding an engineer (#7295)
PostHog/posthog.com: contents/handbook/people/onboarding.md

- Please make sure you spend at least 3 days together, working through the [first week onboarding list](/handbook/people/onboarding#during-their-first-week) and spending time working on any role-specific tasks that are outlined in the new joiner's personal onboarding issue. 
- You will remain the new joiner's main point of contact for the first few weeks, so please continue to check in with them at least once a week for the first month or so. 

> ## In-person onboarding

Reviewer: Are we linking to this from the onboarding GH issues already? If not we likely want to

Usually a new joiner will meet with members of their team in-person to go through the onboarding process. Upon acceptance of an offer, your Team Lead will notify the People & Operations team who will help you coordinate travel if necessary.

In these cases, the process is:

  - Preemptively create the new team member a Google account
  - Issue them a Brex card to their work email with a sufficiently high temporary balance to cover travel costs
  - Have the new team member book travel as usual

### Engineering

We've made dozens of software engineering hires so far at PostHog. This means we know a thing or two about onboarding engineers by now.  

> We got it down to **four rules for successful onboarding of an engineer**:

Author: It's definitely supposed to be a _little_ self-aggrandizing. 😄 They are "rules" for a reason. Rules of thumb, if you will – time in person in precious and we've seen these simple practices to have a positive impact, so we should just always follow them.

> We got it down to **four rules for successful onboarding of an engineer**:

Reviewer 2: Phrasing thing this way just makes me feel like "we know it all, it's done, not changing." Which I don't think is the intent. But up to you! I don't think nit-picking here is going to matter all that much 😄

> We got it down to **four rules for successful onboarding of an engineer**:

Author: That's fair, rewrote a bit to include a point about sharing the knowledge, and de-self-aggrandize a bit

1. _On day one_: You got their development environment all set up and shipped something useful together – even if tiny! It feels great to hit the ground running.

> 2. _By day two_: You've run at least one 1:1 learning session with the new teammate. Give them all the context they need to succeed. (And by the end of the onboarding, each team member should've run at least one such session.)

Reviewer: I'd say more is better ... I'd rather say run 1 per day minimum

Author: Fair point, it seems overly prescriptive to say "by day two". Reworded the points to avoid that, using "every day" instead in this point.

    <details>
      <summary>Looking for learning session ideas?</summary>
      <p>Here's a non-exhaustive list:</p>
      <ul>
        <li>How an event gets from a client library all the way to the Live Events view</li>
        <li>How we turn all our TSX and SCSS files into a fast frontend served from S3</li>
        <li>The architecture of PostHog Cloud</li>
        <li>Query nodes and how they're used throughout the app</li>
        <li>What the dead letter queue is for</li>
        <li>How PostHog experiment results are calculated</li>
        <li>What engineering planning looks like at PostHog</li>
      </ul>
      <p>
        Any of these chats can take as little as 15 minutes or as long as 1 hour, depending on the level of detail.
        You'll also find that some topics apply perfectly in some teams, but not so much in others.
        This is all up to you!
      </p>
    </details>
3. Do at least one brainstorming session on a topic important for the team, writing down actionable conclusions. Use the time together to discuss issues and involve the new joiner in decisions.

> 4. Pair whenever possible. Everyone's sitting right next to each other, so pick work that can benefit from this!

Author: Hmm, maybe this clears it up a bit?
```suggestion
4. Pair whenever possible. You're all sitting next to each other, so pick work that can benefit from in-person collaboration!
```

> 1. Ship something together on day one – even if tiny! It feels great to hit the ground running, with a development environment all ready to go.
> 2. Run 1:1 learning sessions with the new teammate every day. Give them all the context they need to succeed. By the end of the onboarding, each team member present should've run at least one such session.
>     <details>
>       <summary>Looking for learning session ideas?</summary>
>       <p>Here's a non-exhaustive list:</p>
>       <ul>
>         <li>How an event gets from a client library all the way to the Live Events view</li>
>         <li>How we turn all our TSX and SCSS files into a fast frontend served from S3</li>
>         <li>The architecture of PostHog Cloud</li>
>         <li>Query nodes and how they're used throughout the app</li>
>         <li>What the dead letter queue is for</li>
>         <li>How PostHog experiment results are calculated</li>
>         <li>What engineering planning looks like at PostHog</li>
>       </ul>
>       <p>
>         Any of these chats can take as little as 15 minutes or as long as 1 hour, depending on the level of detail.
>         You'll also find that some topics apply perfectly in some teams, but not so much in others.
>         This is all up to you!
>       </p>
>     </details>
> 3. Do at least one brainstorming session on a topic important for the team, writing down actionable conclusions. Use the time together to discuss issues and involve the new joiner in decisions.
> 4. Pair whenever possible. You're all sitting next to each other, so pick work that can benefit from in-person collaboration!

Reviewer 3: ```suggestion
1. Ship something together on day one – even if tiny! It feels great to hit the ground running, with a development environment all ready to go.

2. Run 1:1 learning sessions with the new teammate every day. Give them all the context they need to succeed. By the end of the onboarding, each team member present should've run at least one such session.
    <details>
      <summary>Looking for learning session ideas?</summary>
      <p>Here's a non-exhaustive list:</p>
      <ul>
        <li>How an event gets from a client library all the way to the Live Events view</li>
        <li>How we turn all our TSX and SCSS files into a fast frontend served from S3</li>
        <li>The architecture of PostHog Cloud</li>
        <li>Query nodes and how they're used throughout the app</li>
        <li>What the dead letter queue is for</li>
        <li>How PostHog experiment results are calculated</li>
        <li>What engineering planning looks like at PostHog</li>
      </ul>
      <p>
        Any of these chats can take as little as 15 minutes or as long as 1 hour, depending on the level of detail.
        You'll also find that some topics apply perfectly in some teams, but not so much in others.
        This is all up to you!
      </p>
    </details>
    
3. Do at least one brainstorming session on a topic important for the team, writing down actionable conclusions. Use the time together to discuss issues and involve the new joiner in decisions.

4. Pair whenever possible. You're all sitting next to each other, so pick work that can benefit from in-person collaboration!
```
